Makefile

```diff
@@ -1,89 +1,47 @@
 export GO15VENDOREXPERIMENT := 1
 
-UNAME := $(shell uname)
-ifeq ($(UNAME), Darwin)
-    HASH := md5
-else
-    HASH := md5sum
-endif
-
 all: build manifests
 
-generate: sync
-	find pkg/ -name "*generated*.go" -exec rm {} -f \;
-	./hack/build-go.sh generate ${WHAT}
-	goimports -w -local kubevirt.io cmd/ pkg/ tests/
-	./hack/bootstrap-ginkgo.sh
-	(cd tools/openapispec/ && go build)
-	tools/openapispec/openapispec --dump-api-spec-path api/openapi-spec/swagger.json
+generate:
+	hack/dockerized "hack/glide-checksync.sh && ./hack/generate.sh"
 
-apidocs: generate
-	./hack/gen-swagger-doc/gen-swagger-docs.sh v1 html
+apidocs:
+	hack/dockerized "hack/glide-checksync.sh && ./hack/generate.sh && ./hack/gen-swagger-doc/gen-swagger-docs.sh v1 html"
 
-build: checksync fmt vet compile
+build:
+	hack/dockerized "hack/glide-checksync.sh && ./hack/check.sh && ./hack/build-go.sh install ${WHAT}"
 
 goveralls:
-	./hack/goveralls.sh
-
-compile:
-	./hack/build-go.sh install ${WHAT}
-
-vet:
-	./hack/build-go.sh vet ${WHAT}
-
-fmt:
-	goimports -w -local kubevirt.io cmd/ pkg/ tests/
+	hack/dockerized "hack/glide-checksync.sh && ./hack/check.sh && ./hack/build-go.sh install && TRAVIS_JOB_ID=${TRAVIS_JOB_ID} TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST} TRAVIS_BRANCH=${TRAVIS_BRANCH} ./hack/goveralls.sh"
 
-fmt-bash:
-	shfmt -i 4 -w cluster/ hack/ images/
-
-test: build
-	./hack/build-go.sh test ${WHAT}
+test:
+	hack/dockerized "hack/glide-checksync.sh && ./hack/check.sh && ./hack/build-go.sh install ${WHAT} && ./hack/build-go.sh test ${WHAT}"
 
 functest:
-	./hack/functests.sh
+	hack/functests.sh
 
 clean:
-	./hack/build-go.sh clean ${WHAT}
-	rm _out/ -rf
+	hack/dockerized "./hack/build-go.sh clean ${WHAT} && rm _out/* -rf && rm tools/openapispec/openapispec -rf"
 	rm tools/openapispec/openapispec -rf
 
 distclean: clean
-	find manifests -name "*.yaml" -type f -delete
+	hack/dockerized "rm -rf vendor/ && rm -f .glide.*.hash && glide cc"
 	rm -rf vendor/
-	rm -f .glide.*.hash
-	glide cc
 
 checksync:
-	@test -f .glide.yaml.hash || ${HASH} glide.yaml > .glide.yaml.hash
-	@if [ "`${HASH} glide.yaml`" != "`cat .glide.yaml.hash`" ]; then \
-		glide cc; \
-		glide update --strip-vendor; \
-		${HASH} glide.yaml > .glide.yaml.hash; \
-		${HASH} glide.lock > .glide.lock.hash; \
-	elif [ "`${HASH} glide.lock`" != "`cat .glide.lock.hash`" ]; then \
- 		make sync; \
- 	fi
+	hack/dockerized	hack/glide-checksync.sh
  
 sync:
-	glide install --strip-vendor
-	${HASH} glide.lock > .glide.lock.hash
+	hack/dockerized "glide install --strip-vendor && md5sum glide.lock > .glide.lock.hash"
```

Comment:
I'm fine with this as long as we commit the vendor folder either in this PR or immediately after submitting this pr.